### ISSUE TITLE
Wrong schema generation for composed types

### ISSUE THREAD

**Issue by m-kay** [swagger-api/swagger-core#3030]
When configuring inheritance through allOf property in the `@Schema` annotation, the generated OAS 3 specification is wrong. It is indeed shown corectly in swagger-ui, however when using the specification to generate client code, the generated code is not correct.

With this java classes
```
class Parent {
	private String sharedProperty;
}

@Schema(allOf = Parent.class)
class Child extends Parent {
	private String property;
}
```
the following OAS schemas are generated:
```
...
{
  "components": {
    "schemas": {
      "Parent": {
        "type": "object",
        "properties": {
          "sharedProperty": {
            "type": "string"
          }
        }
      },
      "Child": {
        "type": "object",
        "properties": {
          "property": {
            "type": "string"
          }
        },
        "allOf": [
          {
            "$ref": "#/components/schemas/Parent"
          }
        ]
      }
    }
  }
}
```
According to the [OAS](https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/) correct would be:
```
...
{
  "components": {
    "schemas": {
      "Parent": {
        "type": "object",
        "properties": {
          "sharedProperty": {
            "type": "string"
          }
        }
      },
      "Child": {
        "allOf": [
          {
            "$ref": "#/components/schemas/Parent"
          },
          {
            "type": "object",
            "properties": {
              "property": {
                "type": "string"
              }
            }
          }
        ]
      }
    }
  }
}
```

**Comment by dalewking** [comment]
FYI, in your example you named it baseProperty and in the OAS you switched to sharedProperty

**Comment by dalewking** [comment]
See #3049 which points out that if this we're fixed to generate the correct OAS, that OAS would not be parsed correctly and would end up duplicating the "property" property.

**Comment by dalewking** [comment]
I wonder if the order of items in the allOf matters. You show it with the $ref last. Can you try it with the $ref first?

**Comment by m-kay** [comment]
Yes exactly, refs have to go after the actual object

**Comment by m-kay** [comment]
Sorry I was wrong. Actually ref should be first, otherwise the Child class does not inherit from Parent at all.
See the example https://github.com/m-kay/oasgenerateallofexample, generate the code with `mvn clean install` and compare the generated code in the packages reffirst and reflast.

**Comment by frantuma** [comment]
thanks for reporting this, also duplicates #3197 and #3063; fixed in #3262 (subtypes must be added to parent to be correctly resolved, see #2863 and [related test](https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-core/src/test/java/io/swagger/v3/core/resolving/Ticket3030Test.java#L48)
